Data on OH mesh ents and MLS interplation

Fuad-HH opened the pull request:
Unlike before, the OH field adapter now supports fields located on other mesh entities.
Interpolation headers are added in the `src/interpolation` directory.

Fuad-HH committed:
support rcb partition

Fuad-HH committed:
mesh entity arg added to OH field

Fuad-HH committed:
add mesh_ent_to_int

abhiyan123 committed:
added adj_search.hpp header in MLSInterpolation.hpp file

abhiyan123 committed:
added regular grid linear interpolator and mls interpolator

abhiyan123 committed:
added interpolator subdirectory

abhiyan123 committed:
added the interpolator folder with header files for MLS and linear methods

abhiyan123 committed:
removed hdf5 header

abhiyan123 committed:
removed unnecessary files

Fuad-HH committed:
Merge pull request #1 from abhiyanpaudel/interpolator

Interpolator

Fuad-HH committed:
enforce number of support

searches and adapts radii until all targets get the minimum number of supports and debug queue (counter overflow)

abhiyan123 committed:
can use any dimension and degree of polynomial

abhiyan123 committed:
Merge branch 'mls_generalMLS' into general_mls

abhiyan123 committed:
added test cases

abhiyan123 committed:
moved interpolator folder to pcms

abhiyan123 committed:
modified to incorporate interpolator

abhiyan123 committed:
updated the file

abhiyan123 committed:
deleted unneceesary files

Fuad-HH committed:
variable radius for degas2

abhiyan123 committed:
cleaned up the basis monomial calculation

abhiyan123 committed:
made it general

abhiyan123 committed:
updated the test cases that verifies the monomial basis calculations

abhiyan123 committed:
added unit test case that verifies the basis calculation

abhiyan123 committed:
updated .gitignore file

jacobmerson reviewed (commented):
I'd rather not merge this specific implementation for DAGMC and instead merge the combined version we discussed earlier today that makes use of the parametric coordinates.

If there are other reasons to merge please let me know.

Fuad-HH commented:
> I'd rather not merge this specific implementation for DAGMC and instead merge the combined version we discussed earlier today that makes use of the parametric coordinates.
> 
> If there are other reasons to merge please let me know.

I agree with you.

abhiyan123 committed:
made changes to accomodate general mls procedure

abhiyan123 committed:
compatible with the generalized mls

abhiyan123 committed:
added test case file name

abhiyan123 committed:
created the test case for generalized mls

Fuad-HH committed:
merge general order interpolation changes from abhiyan's fork

Fuad-HH committed:
move include adj_search_dega2.hpp from MLSInterpolation.hpp to test_rbf_interp.cpp

Fuad-HH committed:
radius adaptation for adj_search

abhiyan123 committed:
added Nan Check

abhiyan123 committed:
formatted the files

Fuad-HH committed:
Merge pull request #7 from abhiyanpaudel/general_mls

assertion for nan added for interpolated values and formatted

Fuad-HH committed:
revert clangformatting in CMakeLists.txt and remove const from support.ptr

Fuad-HH committed:
merge centroid->node interp  with general

added some checks too that ends up breaking the code and some debugging
helper functions

Fuad-HH committed:
refactor

removed some unneccesary comments

Fuad-HH committed:
radius adaptation option added for cell2node search

abhiyan123 committed:
renamed test sections

abhiyan123 committed:
interface for passing various rbf functions

abhiyan123 committed:
updated the CMake list to include the new files

abhiyan123 committed:
changes to accomodate interface

abhiyan123 committed:
changes to accomodate interface

abhiyan123 committed:
updated test CMake list

abhiyan123 committed:
added pcms_interpolator target

abhiyan123 committed:
set target properties for pcms_interpolator

abhiyan123 committed:
updated with kokkos kernels LU solver

abhiyan123 committed:
upgraded to kokkos kernels

abhiyan123 committed:
add radial basis function choices for user

abhiyan123 committed:
added and updated test cases

Fuad-HH committed:
Merge pull request #14 from abhiyanpaudel/general_mls

added kokkos kernels LU solver
